backend/src/apiserver/server/util_test.go

```diff
@@ -336,6 +335,92 @@ func TestValidatePipelineSpec_ParameterTooLong(t *testing.T) {
 	assert.Contains(t, err.Error(), "The input parameter length exceed maximum size")
 }
 
+func TestCheckPipelineVersionReference_PipelineVersionReference(t *testing.T) {
```

Comment:
I know usually we don't do UT for private methods. However, if we don't do UT here, we will need to do duplicates UT for both job and run.

Reply:
Discussed offline. The util should be moved to another package and some util methods should be public.